src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/InteractiveRequest.cs

```diff
@@ -179,5 +179,11 @@ private async Task<MsalTokenResponse> RunBrokerWithInstallUriAsync(string broker
             Metrics.IncrementTotalAccessTokensFromBroker();
             return tokenResponse;
         }
+
+        protected override bool ShouldNotCacheTokenResponse()
+        {
+            return AuthenticationRequestParameters.PopAuthenticationConfiguration != null && //limited to pop tokens for the moment. Will be expanded later
+                ServiceBundle.Config.BrokerCreatorFunc != null; //Check to see if MSAL runtime is used.
```

Comment:
not a realiable way to check for WAM. But in any case, having a POP config should be enough.